ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/FinalProjectNode.java

```diff
@@ -104,6 +108,7 @@ public void validateKeyPresent(final SourceName sinkName) {
 
     if (into.isPresent()) {
       // Persistent queries have key columns as key columns - so final projection can exclude them:
```

Comment:
nit (not your code):
```suggestion
      // Persistent queries have key columns as value columns - so final projection can exclude them:
```